pyEpiabm/pyEpiabm/intervention/abstract_intervention.py

```diff
@@ -0,0 +1,62 @@
+#
+# AbstractIntervention Class
+#
+
+
+class AbstractIntervention:
+    """Abstract class for Interventions.
+
+    """
+    def __init__(self, start_time, policy_duration, population,
+                 case_threshold=0, **kwargs):
+        """Set the parameters of the interventions.
+
+        Parameters
+        ----------
+        start_time : float
+            Start time of intervention
+        policy_duration : float
+            Duration of the intervention
+        case_threshold : float
+            Number of cases required to trigger the intervention
+        population : Population
+            Population: :class:`Population` to bind
+
+        """
+        self.start_time = start_time
+        self.policy_duration = policy_duration
+        self.case_threshold = case_threshold
+        self._population = population
+
+    def is_active(self, time, num_cases):
+        """Query if the intervention is currently active.
+
+        Parameters
+        ----------
+        time : float
+            Current simulation time
+        num_cases : integer
```

Comment:
```suggestion
        num_cases : int
```

This type in python is just called `int`